compiler-core/src/type_/hydrator.rs

```diff
@@ -121,12 +121,23 @@ impl Hydrator {
                 let TypeConstructor {
                     parameters,
                     typ: return_type,
+                    deprecation,
                     ..
                 } = environment
                     .get_type_constructor(module, name, *location)
                     .map_err(|e| convert_get_type_constructor_error(e, location))?
                     .clone();
 
+                match deprecation {
+                    Deprecation::NotDeprecated => {}
+                    Deprecation::Deprecated { message } => {
+                        environment.warnings.emit(Warning::DeprecatedValue {
```

Comment:
This results in a warning saying that the _value_ is deprecated, but it's a type. Could you make a `DeprecatedType` variant or make this `DeprecatedItem` and have it hold a `Layer` to indicate whether it's a value or a type.

Reply:
Oh Yeah. Will try making a DeprecatedItem with a Layer.